src/Datadog.Trace/Sampling/RegexSamplingRule.cs

```diff
@@ -0,0 +1,178 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Sampling
+{
+    internal class RegexSamplingRule : ISamplingRule
+    {
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<RegexSamplingRule>();
+        private static readonly TimeSpan RegexTimeout = TimeSpan.FromMilliseconds(500);
+
+        private readonly float _samplingRate;
+        private readonly string _serviceNameRegex;
+        private readonly string _operationNameRegex;
+
+        private bool _hasPoisonedRegex = false;
```

Comment:
Just wondering whether or not it is too much to try to remove the poisoned regex from the sampling rules. Or maybe it doesn't always come from the `RuleBasedSampler's internal `_rules` list?

Reply:
I could, seemed like a bigger change for a very rare case though.

Reply:
And yes, *theoretically* there can be multiple Tracers/Samplers. So *theoretically* this one rule could be used in multiple places. 
That would be a bit silly though.

Reply:
yeah, you would have to weigh that against having another piece of data in the object. I'm ok with the private bool, as long as it is thread safe which I doubt is a matter here.